### DISCUSSION TITLE
posts/2022-05-27-youmaynotneedabundler

### DISCUSSION THREAD

**Discussion by giscus** [cmdcolin/cmdcolin.github.io#10] [General]
# posts/2022-05-27-youmaynotneedabundler

Blogging for the future

https://cmdcolin.github.io/posts/2022-05-27-youmaynotneedabundler

**Comment by billydarmawan** [comment]
what would you recommend if I need to add css files? does that mean I finally need a bundler and cannot stick with `tsc` only?

**Reply by cmdcolin** [reply]
If you want your library to add css and automatically add it to the document, you can manually write code that runs stuff that manually adds it to the document in a style tag. the bundler that allows using css likely does a similar thing under the hood e.g. esbuild css plugins, roll up css plugins, etc.

```
  const style = document.createElement('style')
  style.append(yourcssstringhere)
  document.head.append(style)
```

or you can instruct users to import a CSS from your library, this is also common

```
import 'yourpackage/thing.css'

```

**Reply by billydarmawan** [reply]
I see, so just to clarify, another option for me is to just publish the CSS files to the `dist` directory and expect that library users will import my CSS file from the `dist`, is it correct?

**Reply by cmdcolin** [reply]
yes that is also allowed:)

**Reply by billydarmawan** [reply]
thanks Colin! your post and answers helps me a lot.

**Comment by vladpuz** [comment]
I also came to the same opinion as the author a few years ago. The feeling that the world had gone crazy did not leave me. 👍
